chef/automate
# Syncing changes to the user menu display name

### :nut_and_bolt: Description: What code changed, and why?
When a user changes their user display name from the profile page or setting-> users page, the display name is updated in the menu.

This one simple bug needed a lot of changes to complete. These changes add the concept of the logged-in user to the UI. A new section to the NGRX store was created that stores information about this user. The logged-in user's ID is first placed in the store from the chef-seesion.service. Where the ID is updated by each session refresh. 

The user-details component has been updated to have three different states: admin editing other users, profile page, and admin editing their own user. The admin editing other users has not changed from before. The profile page is different in that it updates the newly created logged-in user NGRX store. The admin editing their own user is different in that when a user tries to update their password they must enter the previous password. This also updates the logged-in NGRX store. 

The main change is to the profile component that displays the logged-in user's information in the top-right menu. Before the user's display name was pulled from the session when they first logged in. The problem was that it could only be changed by logging out and back in. Now, the display name is pulled from the newly created logged-in NGRX store. Therefore when changes to the display name are made the menu updates. 

A flag to say if the user is a local or non-local user was also added. For non-local users (for example SAML) they do not have a configurable display name through Automate. Because of this, we have to remove the profile page link and not try to pull the logged-in user's information from the APIs.  To figure out if a user is local or not we are looking at the federated claims field on the session ID token. The federated claims have a field called connector_id that has the term 'local' for local users and 'saml' for non-local users. 

### :chains: Related Resources
https://github.com/chef/automate/issues/2660

### :+1: Definition of Done
* Change display name from the user profile page
* Change display name from the setting -> user page
* Change the logged-in user's password from the setting -> user page needs the user's previous password

### :athletic_shoe: How to Build and Test the Change
1. Start the UI
1. Go to the user profile page and update the display name
1. Ensure the menu name is updated. 
1. Go to the logged-in user's page from the settings -> users page and update the display name. 
1. Ensure the menu name is updated. 
1. Go to the logged-in user's page from the settings -> users page and update the password. 
1. Ensure the previous password is needed. 

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
